### PR TITLE
Feature/hide pieces and movements

### PR DESCRIPTION
**Pull request by vee-L**
Everything works except removing the hidden elements from suggestions (which I spent a ridiculous amount of time on for no real results, so I figured it was best to move forward).

### CODE REVIEW

File: elvis/settings.py [hunk 1]
```diff
@@ -162,15 +162,15 @@
...
 
-if SETTING_TYPE != "local":
-    DATABASES = {
-        'default': {
-            'ENGINE': 'django.db.backends.postgresql_psycopg2',
-            'NAME': 'elvis_{0}'.format(SETTING_TYPE),
-            'USER': 'elvisdatabase',
-            'PASSWORD': '5115C67O2v3GN31T49Md'
-            }
+#if SETTING_TYPE != "local":
```
Comment (ahankinson) [inline elvis/settings.py]: Is this correct? Will commenting out this line break anything?

File: elvis/settings.py [hunk 2]
```diff
@@ -282,7 +282,8 @@
         'file': {
             'level': 'DEBUG',
             'class': 'logging.FileHandler',
-            'filename': '/var/log/elvisdb/{0}/django_elvis.log'.format(SETTING_TYPE)
+            #'filename': '/srv/webapps/elvisdb_logs/{0}/django_elvis.log'.format(SETTING_TYPE)
+            'filename': '/Users/veroniquelagace/Documents/code/elvis/elvis-database/logs/test.log'.format(SETTING_TYPE)
```
Comment (ahankinson) [inline elvis/settings.py]: This probably shouldn't be changed in production.

File: elvis/models/elvis_model.py [hunk 3]
```diff
@@ -123,8 +125,14 @@ def save(self, force_insert=False, force_update=False, using=None,
                 a.auto_rename(**kwargs)
             for m in self.movements.all():
                 m.save(**kwargs)
+            if self.hidden:
+                self.solr_delete(commit=True);
 
-        if kwargs.get("ignore_solr"):
+        if (cls=="Movement" or cls=="Piece"):
```
Comment (ahankinson) [inline elvis/models/elvis_model.py]: spaces between `==` and the condition

File: elvis/views/views.py [hunk 4]
```diff
@@ -88,9 +88,14 @@ def solr_suggest(request):
                 for i in range(min(7, len(sorted_suggestions))):
                     results.append({'name': sorted_suggestions[i]['term']})
         else:
-            url = settings.SOLR_SERVER + "/suggest/?wt=json" \
-                                         "&suggest.dictionary={0}" \
-                                         "&q={1}".format(dictionary, value)
+            if dictionary == "pieceSuggest":
```
Comment (ahankinson) [inline elvis/views/views.py]: These two are exactly the same.

File: elvis/models/elvis_model.py [hunk 5]
```diff
@@ -123,6 +125,8 @@ def save(self, force_insert=False, force_update=False, using=None,
                 a.auto_rename(**kwargs)
             for m in self.movements.all():
                 m.save(**kwargs)
+            # if self.hidden:
```
Comment (lexpar) [inline elvis/models/elvis_model.py]: Can these be removed?

File: elvis/serializers/serializers.py [hunk 6]
```diff
@@ -127,10 +130,13 @@ def _permission_appender(self, result, instance):
             perms = {'can_edit': True, 'can_view': True}
         elif not instance.__dict__.get('public', True):
             perms = {'can_edit': False, 'can_view': False}
+        elif (instance.__class__.__name__ == "Piece" or instance.__class__.__name__ == "Movement") and instance.__dict__.get('hidden', True):
```
Comment (lexpar) [inline elvis/serializers/serializers.py]: I believe `elif instance.__dict__.get('hidden', False)` would be equivalent to this line.

File: elvis/views/common.py [hunk 7]
```diff
@@ -28,6 +28,8 @@ def determine_perms(self, request, *args, **kwargs):
 
         if not obj.__dict__.get('public', True):
             return {'can_edit': False, 'can_view': False}
+        elif obj.__dict__.get('hidden', True):
```
Comment (lexpar) [inline elvis/views/common.py]: This means that if 'hidden' is undefined on the dict, then the item defaults to being hidden. Is that how it's supposed to work?

**Pull request merged**